server/sonar-server/src/test/java/org/sonar/server/computation/step/PurgeRemovedViewsStepTest.java

```diff
@@ -0,0 +1,126 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.computation.step;
+
+import com.google.common.base.Function;
+import com.google.common.collect.Iterables;
+import org.elasticsearch.search.SearchHit;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.sonar.api.config.Settings;
+import org.sonar.api.resources.Qualifiers;
+import org.sonar.core.component.ComponentDto;
+import org.sonar.core.persistence.DbSession;
+import org.sonar.core.persistence.DbTester;
+import org.sonar.server.component.ComponentTesting;
+import org.sonar.server.component.db.ComponentDao;
+import org.sonar.server.computation.ComputationContext;
+import org.sonar.server.db.DbClient;
+import org.sonar.server.es.EsTester;
+import org.sonar.server.issue.db.IssueDao;
+import org.sonar.server.view.index.ViewDoc;
+import org.sonar.server.view.index.ViewIndex;
+import org.sonar.server.view.index.ViewIndexDefinition;
+
+import java.util.List;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class PurgeRemovedViewsStepTest {
+
+  @Rule
+  public EsTester esTester = new EsTester().addDefinitions(new ViewIndexDefinition(new Settings()));
+
+  @Rule
+  public DbTester db = new DbTester();
+
+  ComputationContext context;
+
+  DbSession session;
+
+  DbClient dbClient;
+
+  PurgeRemovedViewsStep step;
+
+  @Before
+  public void setUp() {
+    context = mock(ComputationContext.class);
+    session = db.myBatis().openSession(false);
+    dbClient = new DbClient(db.database(), db.myBatis(), new IssueDao(db.myBatis()), new ComponentDao());
+    step = new PurgeRemovedViewsStep(new ViewIndex(esTester.client()), dbClient);
+  }
+
+  @After
+  public void after() {
+    this.session.close();
+  }
+
+  @Test
+  public void purge_removed_views() throws Exception {
+    when(context.getProject()).thenReturn(ComponentTesting.newProjectDto("DBCA").setQualifier(Qualifiers.VIEW));
+
+    esTester.putDocuments(ViewIndexDefinition.INDEX, ViewIndexDefinition.TYPE_VIEW,
+      new ViewDoc().setUuid("ABCD").getFields(),
+      new ViewDoc().setUuid("BCDE").getFields(),
+      // Should be removed as it no more exists in db
+      new ViewDoc().setUuid("CDEF").getFields());
+
+    ComponentDto view = ComponentTesting.newProjectDto("ABCD").setQualifier(Qualifiers.VIEW);
+    ComponentDto subView = ComponentTesting.newModuleDto(view, "BCDE").setQualifier(Qualifiers.SUBVIEW);
+    dbClient.componentDao().insert(session, view, subView);
+    session.commit();
+
+    step.execute(context);
+
+    List<SearchHit> results = esTester.getDocuments(ViewIndexDefinition.INDEX, ViewIndexDefinition.TYPE_VIEW);
+    List<String> viewUuids = newArrayList(Iterables.transform(results, new Function<SearchHit, String>() {
```

Comment:
This function is helpful and must be moved to EsTester:
```java
public <T> Iterable<T> getDocumentFields(String indexName, String typeName, final String fieldName) {
    return Iterables.transform(getDocuments(indexName, typeName), new Function<SearchHit, T>() {
      @Override
      public T apply(SearchHit input) {
        return (T)input.sourceAsMap().get(fieldName);
      }
    });
  }
```

Test would be simply: 
```java
Iterable<String> viewUuids = esTester.getDocumentFields(ViewIndexDefinition.INDEX, ViewIndexDefinition.TYPE_VIEW, ViewIndexDefinition.FIELD_UUID);
assertThat(viewUuids).containsOnly("ABCD", "BCDE");
```